core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala

```diff
@@ -236,13 +265,25 @@ class HadoopMapReduceCommitProtocol(
     committer.setupTask(taskContext)
     addedAbsPathFiles = mutable.Map[String, String]()
     partitionPaths = mutable.Set[String]()
+    dynamicStagingTaskFiles = mutable.Set[Path]()
   }
 
   override def commitTask(taskContext: TaskAttemptContext): TaskCommitMessage = {
     val attemptId = taskContext.getTaskAttemptID
     logTrace(s"Commit task ${attemptId}")
     SparkHadoopMapRedUtil.commitTask(
       committer, taskContext, attemptId.getJobID.getId, attemptId.getTaskID.getId)
+    if (dynamicPartitionOverwrite) {
```

Comment:
so where does SparkHadoopMapRedUtil.commitTask moves the data from/to and how does dynamicPartitionOverwrite deal with that output? Can you please give an example? It seems there are 2 renames happening one due to SparkHadoopMapRedUtil.commitTask and then with the code block inside dynamicPartitionOverwrite.

Reply:
IIUC, `SparkHadoopMapRedUtil.commitTask` will output to `dynamicStagingTaskFiles` where each task attempt gets its own staging directory to avoid conflicts. Then, we rename the staging files to their final destination if not exist. 

One minor issue is the following log in `SparkHadoopMapRedUtil.commitTask`. When we say the task attempt has committed, it's actually not. @turboFei 

```        
logInfo(s"$mrTaskAttemptID: Committed")
```

Reply:
yeah, this log is a little confused.